o2a/o2a_libs/el_parser.py

```diff
@@ -0,0 +1,296 @@
+        operator.value = "<="
+
+    if operator.value == "ne":
+        operator.value = "!="
+
+    if operator.value == "eq":
+        operator.value = "=="
+
+    if operator.value == "||":
+        operator.value = "or"
+
+    if operator.value == "&&":
+        operator.value = "and"
+
+    if operator.value == "mod":
+        operator.value = "%"
+
+    if operator.value == "div":
+        operator.value = "/"
+
+    operator.value = " " + operator.value + " "
+
+    tree.children = [operator]
+    return tree
+
+
+def _translate_ternary(tree: Tree, case: int) -> str:
+    """
+    Translates ternary expression.
+    """
+    if case == 1:
+        # f() ? true : false
+        condition, ternary = tree.children
+        _, if_true, _, if_false = ternary.children
+        translation = (
+            f"{_translate_el(if_true)} if " f"{_translate_el(condition)} else {_translate_el(if_false)}"
+        )
+        return translation
+
+    if case == 2:
+        # x op y ? true : false
+        first = _translate_el(tree.children[0])
+        operator = _translate_el(tree.children[1])
+
+        # expression | ternary
+        expression, ternary = tree.children[2].children
+
+        # ? | expression | : | expression
+        _, if_true, _, if_false = ternary.children
+
+        second = _translate_el(expression)
+
+        condition = f"{first}{operator}{second}"
+        translation = f"{_translate_el(if_true)} if {condition} else {_translate_el(if_false)}"
+        return translation
+
+    raise Exception("Unhandled ternary expression.")
+
+
+def _translate_function(tree: Tree) -> Tree:
+    """
+    Translates function invocations.
+    """
+    new_children = []
+    for child in tree.children:
+        if isinstance(child, Token) and child.value == ":":
+            child.value = "_"
+        new_children.append(child)
+
+    tree.children = new_children
+    return tree
+
+
+def _translate_token(token: Token) -> str:
+    """
+    Translates non-python values to python equivalents.
+    """
+    if token.value in ("${", "#{"):
+        token.value = "{{"
+
+    if token.value == "}":
+        token.value = "}}"
+
+    if token.type == "NULL":
+        token.value = None
+
+    if token.type == "BOOL":
+        if token.value == "true":
+            token.value = True
+        token.value = False
+
+    if token.type == "JAVA":
+        token.value = _camel_to_snake(token.value)
+
+    return str(token.value)
+
+
+def _translate_el(tree: Union[Tree, Token]) -> str:
+    """
+    Translates el expression to jinjia equivalent.
+    """
+    output = ""
+
+    if isinstance(tree, Token):
+        return _translate_token(tree)
+
+    if tree.data == "function_invocation":
+        tree = _translate_function(tree)
+
+    if tree.data == "binary_op":
+        tree = _translate_binary_operator(tree)
+
+    if tree.data == "expression" and "ternary" in [ch.data for ch in tree.children]:
+        return _translate_ternary(tree, case=1)
```

Comment:
nit: case ={1,2} isn't very readable.

Reply:
Agreed! I think it would be best to move this logic to `_translate_ternary()`.